app/facades/home_facade.rb

```diff
@@ -0,0 +1,24 @@
+class HomeFacade
+  attr_reader :product
+
+  def initialize(product)
+    @product = product
+    @images = @product.images.to_a
+  end
+
+  def image_main
+    @images.find do |image|
+      image.tag == Image::TAG[:main]
+    end
+  end
+
+  def images_normal
+    @images.select do |image|
+      image.tag != Image::TAG[:main]
+    end
+  end
+
+  def relead_products
+    Product.where(category_id: @product.category_id).limit(3)
```

Comment:
all other methods can access `product` directly (thanks to `attr_reader`), no need to use `@product`.